src/Apps/Order2/Routes/Checkout/Order2CheckoutApp.tsx

```diff
@@ -0,0 +1,111 @@
+import { Column, Flex, GridColumns, Stack, Text } from "@artsy/palette"
+import { CheckoutStepName } from "Apps/Order2/Routes/Checkout/CheckoutContext/utils"
+import { Order2CheckoutLoadingSkeleton } from "Apps/Order2/Routes/Checkout/Components/Order2CheckoutLoadingSkeleton"
+import { Order2CollapsibleOrderSummary } from "Apps/Order2/Routes/Checkout/Components/Order2CollapsibleOrderSummary"
+import { Order2FulfillmentDetailsStep } from "Apps/Order2/Routes/Checkout/Components/Order2FulfillmentDetailsStep/Order2FulfillmentDetailsStep"
+import { Order2ReviewStep } from "Apps/Order2/Routes/Checkout/Components/Order2ReviewStep"
+import { useCheckoutContext } from "Apps/Order2/Routes/Checkout/Hooks/useCheckoutContext"
+import { ErrorPage } from "Components/ErrorPage"
+import { useSystemContext } from "System/Hooks/useSystemContext"
+import type { Order2CheckoutApp_viewer$key } from "__generated__/Order2CheckoutApp_viewer.graphql"
+import { Meta, Title } from "react-head"
+import { graphql, useFragment } from "react-relay"
+interface Order2CheckoutAppProps {
+  viewer: Order2CheckoutApp_viewer$key
+}
+
+export const Order2CheckoutApp: React.FC<Order2CheckoutAppProps> = ({
+  viewer,
+}) => {
+  const { isEigen } = useSystemContext()
+
+  const data = useFragment(FRAGMENT, viewer)
+  const order = data.me?.order
+
+  // TODO: CTX is always present, but the hook needs to be updated after we can guarantee
+  // this from its usage in the shared express checkout
+  const { isLoading, steps } = useCheckoutContext()!
+
+  const stepNames = steps?.map(step => step.name)
+
+  if (!order) {
+    return <ErrorPage code={404} message="Order not found" />
+  }
+
+  return (
+    <>
+      <Title>Checkout | Artsy</Title>
+      <Meta
+        name="viewport"
+        content={
+          isEigen
+            ? "width=device-width, user-scalable=no"
+            : "width=device-width, initial-scale=1, maximum-scale=5 viewport-fit=cover"
+        }
+      />
+      {isLoading && <Order2CheckoutLoadingSkeleton order={order} />}
+      <GridColumns style={{ display: isLoading ? "none" : "block" }}>
+        <Column span={[12, 8]} start={[1, 2]}>
+          <Stack gap={1} bg="mono5">
+            {/* Collapsible order summary */}
+            <Order2CollapsibleOrderSummary order={order} />
+
+            {/* Fulfillment details Step */}
+            <Order2FulfillmentDetailsStep order={order} />
+
+            {/* Shipping method Step */}
+            {stepNames.includes(CheckoutStepName.DELIVERY_OPTION) && (
+              <Flex flexDirection="column" backgroundColor="mono0" p={2}>
```

Comment:
This is a small behavior change - if pickup is set then this section will disappear.

There is no other dependency on the context `steps` for deciding whether to render a step here because they are always required, but we could do that. Technically the ordering of the steps in this file needs to match the ordering of the array of steps.